tests/integration/scripts/test_validate_integration.py

```diff
@@ -94,3 +94,64 @@ async def test_validate_against_source_gets_old_tvars(
         flow_file.read_text().replace('P1Y = foo', 'P1Y = {{FOO}}'))
     with pytest.raises(Jinja2Error):
         validate(src_dir)
+
+
+@pytest.mark.compatmode
+def test_validate_simple_graph(flow, validate, caplog):
+    """Test deprecation notice for Cylc 7 simple graph (no recurrence section)
+    """
+    id_ = flow({
+        'scheduler': {'allow implicit tasks': True},
+        'scheduling': {'dependencies': {'graph': 'foo'}}
+    })
+    validate(id_)
+    expect = (
+        'deprecated graph items were automatically upgraded '
+        'in "workflow definition":'
+        '\n * (8.0.0) [scheduling][dependencies][X]graph'
+        ' -> [scheduling][graph]X - for X in:\n       graph'
+    )
+    assert expect in caplog.messages
+
+
+@pytest.mark.parametrize(
```

Comment:
I don't think this is a valid use case for parametrization, the test only needs to be run once, this runs it once per message to check.

Use a for loop instead:

```python
python
validate(...)
for message in [
    ...
]:
    assert message in ...
```

Note, if a comparison fails, pytest will print the difference, so it's easy to tell which message the test fails on.